app/queries/users_by_thread_volume_query.rb

```diff
@@ -0,0 +1,22 @@
+class UsersByThreadVolumeQuery
```

Comment:
rails forces its naming conventions hard. moving queries into app means you namespace on 2nd level depth. So Queries:: is basically wrong.. and it has to live in app/extras/queries just to work as is.